### ISSUE TITLE
SKT:Training problem of SKT model

### ISSUE THREAD

**Issue by Dajcaac** [bigdata-ustc/XKT#15]
Hello, excuse me. I encountered some problems in the process of reproducing SKT. When I used the Assistment2015 data set, the learning rate and batch_size were the same as in the paper (lr=0.001, bs=16), but the AUC(macro_auc) of the training result was only 0.62. Is there anything I should pay attention to during training? Why my AUC results are so bad. Also, could you please explain the meaning of "macro_aupoc"? I don't quite understand what it means. Finally, I look forward to your reply. Thank you very much

**Comment by Dajcaac** [comment]
`logger: <Logger model (INFO)>
model_name: model
model_dir: model
begin_epoch: 0
end_epoch: 30
batch_size: 16
save_epoch: 1
optimizer: Adam
optimizer_params: {'learning_rate': 0.001, 'wd': 0.001, 'clip_gradient': 1}
lr_params: {}
train_select: None
save_select: None
ctx: cpu(0)
train_ctx: None
eval_ctx: None
toolbox_params: {}
hyper_params: {'ku_num': 101, 'graph_params': [['../../data/assistment_2015/correct_transition_graph.json', True], ['../../data/assistment_2015/ctrans_sim.json', False]], 'hidden_num': 5}
init_params: {}
loss_params: {}
caption: 
validation_result_file: model\result.json
cfg_path: model\configuration.json
Epoch| Total-E          Batch     Total-B       Loss-SequenceLogisticMaskLoss             Progress           
    0|      29           1034        1034                            0.237072     [11:25<00:00, 1.51it/s]    
evaluating: 100%|██████████| 314/314 [01:48<00:00,  2.88it/s]
D:\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
D:\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
D:\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
Epoch [0] Loss - SequenceLogisticMaskLoss: 0.237072
                  precision        recall        f1  support
0.0               0.000000     0.0  0.000000    41280
1.0               0.708139     1.0  0.829135   100157
macro_avg   0.354069     0.5  0.414567   141437
accuracy: 0.708139 macro_auc: 0.520091 macro_aupoc: 0.714649

**Comment by Dajcaac** [comment]
logger: <Logger model (INFO)>
model_name: model
model_dir: model
begin_epoch: 0
end_epoch: 30
batch_size: 16
save_epoch: 1
optimizer: Adam
optimizer_params: {'learning_rate': 0.001, 'wd': 0.001, 'clip_gradient': 1}
lr_params: {}
train_select: None
save_select: None
ctx: cpu(0)
train_ctx: None
eval_ctx: None
toolbox_params: {}
hyper_params: {'ku_num': 101, 'graph_params': [['../../data/assistment_2015/correct_transition_graph.json', True], ['../../data/assistment_2015/ctrans_sim.json', False]], 'hidden_num': 5}
init_params: {}
loss_params: {}
caption: 
validation_result_file: model\result.json
cfg_path: model\configuration.json
Epoch| Total-E          Batch     Total-B       Loss-SequenceLogisticMaskLoss             Progress           
    0|      29           1034        1034                            0.237072     [11:25<00:00, 1.51it/s]    
evaluating: 100%|██████████| 314/314 [01:48<00:00,  2.88it/s]
D:\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
D:\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
D:\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
Epoch [0] Loss - SequenceLogisticMaskLoss: 0.237072
                    precision    recall        f1       support
0.0               0.000000     0.0  0.000000    41280
1.0               0.708139     1.0  0.829135   100157
macro_avg   0.354069     0.5  0.414567   141437
accuracy:   0.708139 macro_auc: 0.520091 macro_aupoc: 0.714649

**Comment by TRRtrr** [comment]
你好，我也想去复现SKT，但是我复现assist2015的先决图和相似图的时候，不能得到论文中汇报的两种边的条数。因此我想问问您，您是如何复现这两种图的呢？能否教教我

**Comment by XDZxdz1** [comment]
Hello, I also want to reproduce SKT, but when I reproduce the prerequisite graph and similar graph of assist2015, I cannot get the number of two edges reported in the paper. So I want to ask you, how do you reproduce these two kinds of graphs? Can you teach me

**Comment by Dajcaac** [comment]
> Hello, I also want to reproduce SKT, but when I reproduce the prerequisite graph and similar graph of assist2015, I cannot get the number of two edges reported in the paper. So I want to ask you, how do you reproduce these two kinds of graphs? Can you teach me



> Hello, I also want to reproduce SKT, but when I reproduce the prerequisite graph and similar graph of assist2015, I cannot get the number of two edges reported in the paper. So I want to ask you, how do you reproduce these two kinds of graphs? Can you teach me

I'm afraid I can't help you, because the number of edges I get is not the same as in the paper

**Comment by Dajcaac** [comment]
> Hello, I also want to reproduce SKT, but when I reproduce the prerequisite graph and similar graph of assist2015, I cannot get the number of two edges reported in the paper. So I want to ask you, how do you reproduce these two kinds of graphs? Can you teach me

I'm afraid I can't help you, because the number of edges I get is not the same as in the paper

**Comment by TRRtrr** [comment]
Hello, excuse me, do you generate the prerequisite edges and similar edges according to the description in the paper

**Comment by TRRtrr** [comment]
I also want to ask you whether this SKT code can only run on the CPU?  If I want to transfer to GPU for operation, What should I do if I want to transfer to GPU for running?
macro_aupoc: I see its calculation method in the code. The path is 
XKT-master\Lib\sitepackages\longling\ML\metrics\classification.py

**Comment by XDZxdz1** [comment]
I also want to ask you whether this SKT code can only run on the CPU? What should I do if I want to transfer to GPU for running?
macro_aupoc: I see its calculation method in the code. The path is
XKT-master\Lib\sitepackages\longling\ML\metrics\classification.py

**Comment by Dajcaac** [comment]
> I also want to ask you whether this SKT code can only run on the CPU? What should I do if I want to transfer to GPU for running? macro_aupoc: I see its calculation method in the code. The path is XKT-master\Lib\sitepackages\longling\ML\metrics\classification.py

You can download the GPU version of mxnet and change the ctx parameter to gpu()
